zerver/lib/export.py

```diff
@@ -1312,12 +1317,7 @@ def get_ids(records: List[Record]) -> Set[int]:
             recipient__in=recipient_ids_for_them,
         )
 
-        message_queries = [
-            messages_we_received,
-            messages_we_sent_to_them,
-        ]
-        if consent_message_id is not None:
-            message_queries.append(messages_we_received_in_protected_history_streams)
```

Comment:
I'm not completely sure this refactor is really an improvement, but it might so I pushed it and we can just drop it if appropriate

Reply:
I think it's an improvement!  I was actually thinking about it when I was just looking at the first commit. :)

I like that it removes having to track the cotext of what sets `messages_we_received_in_protected_history_streams`, and that it's not always defined.